File: my/MCPClient.py
from ast import arg
import asyncio
import json
# from model_context_protocol.client import Client\
from mcp import client
# from mcp.client.stdio import 
from mcp import ClientSession, StdioServerParameters, Tool
from mcp.client.stdio import stdio_client
# from mcp.types import Tool

from typing import Optional
from contextlib import AsyncExitStack
# from anthropic import Anthropic
from dotenv import load_dotenv

load_dotenv() # load environment variables from .env

class MCPClient:
    def __init__(
        self, 
        name: str,
        command: str,
        args: list[str],
        version: str = "1.0.0"
    ):
        # self.mcp = client(name=name, version=version)
        self.session: 
        self.transport = None
        self.tools: list[Tool] = []
        self.command = command
        self.args = args

    async def close(self):
        """清理资源，关闭与服务器的连接"""
        await self.mcp.

    async def init(self):
        """公开的初始化方法，启动服务器连接"""
        await self.__connect_to_server()

File: my/MCPClient.ts
import { Client } from "@modelcontextprotocol/sdk/client/index.js";
import { StdioClientTransport } from "@modelcontextprotocol/sdk/client/stdio.js";
import { SetLevelRequest, Tool } from "@modelcontextprotocol/sdk/types.js";

export default class MCPClient{
    private mcp: Client;
    private transport: StdioClientTransport | null = null;
    private tools: Tool[] = []
    private command: string
    private args: string[]

    constructor(name: string, command: string, args: string[], version?: string){
        this.mcp = new Client({ name, version: version || '1.0.0'})
        this.command = command
        this.args = args
    }

    public async close(){
        await this.mcp.close()
    }
    public async init(){
        await this.connectToServer()
    }
    public gettools(){
        return this.tools
    }   
    public async callTool(name: string, params: Record<string, any>){
        return await this.mcp.callTool({name, arguments: params})
    }

    private async connectToServer() {
        try {
            this.transport = new StdioClientTransport({
                command: this.command,
                args: this.args,
            });
            //避免初始化之前就调用mcp功能，这里return的是一个promise
            await this.mcp.connect(this.transport);

            const toolsResult = await this.mcp.listTools();
            this.tools = toolsResult.tools.map((tool) => {
                return {
                    name: tool.name,
                    description: tool.description,
                    inputSchema: tool.inputSchema,
                };
            });
            console.log(
            "Connected to server with tools:",
            this.tools.map(({ name }) => name)
            );
        } catch (e) {
            console.log("Failed to connect to MCP server: ", e);
            throw e;
        }
}
}

